### ISSUE TITLE
3.0.0 migration docs inaccuracy

### ISSUE THREAD

**Issue by rachel-bousfield** [wasmerio/wasmer#3441]
Right [here](https://github.com/wasmerio/wasmer/blob/master/docs/migration_to_3.0.0.md#after) in the 2.3 to 3.0.0 migration docs we have the following excerpt
```rs
pub struct MyEnv {
    pub memory: Option<Memory>,
    pub alloc_guest_memory: Option<TypedFunction<i32, i32>>,
    pub multiply_by: u32,
}

let env = FunctionEnv::new(&mut store, MyEnv {
  memory: None,
  alloc_guest_memory: None,
  multiply_by: 10,
});

let instance = Instance::new(&module, &imports);
let mut env_mut = env.as_mut(&mut store);
env_mut.memory = Some(instance.exports.get_memory("memory"));
env_mut.alloc_guest_memory = Some(instance.exports.get_typed_function("__alloc"));
```

Observe that `Instance::new`, however, requires a `&mut store` as the first argument.
Unfortunately doing so now means two mutable references to `store` are alive at the same time, so the example will have to be reworked beyond the straightforward
```rs
let instance = Instance::new(&mut store, &module, &imports);
```

**Comment by syrusakbary** [comment]
That's a great catch. We need to fix this

**Comment by w-flo** [comment]
Also, this sets the `memory: Option<Memory>` field in the MyEnv struct to `Some(instance.exports.get_memory("memory"))`, which is a `Result<>`, so something like unwrap is needed. Even after the unwrap, it's `&Memory` instead of `Memory` so that won't work.

I came to this bug report to find out how to access linear memory in a host function, and it looks like I need a different approach. :-)

**Comment by ptitSeb** [comment]
Fixed in 3.2.1.
